include/odp/api/spec/packet_io.h

```diff
@@ -970,6 +970,36 @@ int odp_pktio_promisc_mode_set(odp_pktio_t pktio, odp_bool_t enable);
 int odp_pktio_promisc_mode(odp_pktio_t pktio);
 
 /**
+ * Maximum frame length at packet input
+ *
+ * Maximum frame length in bytes that the packet IO interface can receive.
+ * For Ethernet, the frame length bytes start with MAC addresses and continue
+ * to the end of the payload. So, Ethernet checksum, interpacket gap
+ * and preamble bytes are excluded from the length.
+ *
+ * @param pktio  Packet IO handle.
+ *
+ * @return Maximum frame length at packet input
+ * @retval 0 on failure
+ */
+uint32_t odp_pktin_maxlen(odp_pktio_t pktio);
```

Comment:
bad naming. odp_pktin_xxx functions take argument odp_pktin_queue_t not odp_pktio_t.  If you want to keep odp_pktio_t as argument for function, then odp_pktio_capability_t is more logical place for it,

Reply:
int odp_pktin_event_queue(odp_pktio_t pktio, odp_queue_t queues[], int num);
int odp_pktin_queue(odp_pktio_t pktio, odp_pktin_queue_t queues[], int num);
uint64_t odp_pktin_wait_time(uint64_t nsec);
uint64_t odp_pktin_ts_res(odp_pktio_t pktio);
odp_time_t odp_pktin_ts_from_ns(odp_pktio_t pktio, uint64_t ns);

These are not the first pktin/pkout functions that have pktio as param. Previously, I considered  odp_pktio_rx_xxx() and odp_pktio_tx_xxx() prefixes for these, but odp_pktin_xxx() and odp_pktout_xxx() was more compact and as intuitive.

Also these are not capabilities but the current configured frame lengths. Later on, we'll add configuration options for changing these and capabilities to indicate min/max values. This is now a  simple replacement of MTU function.

Reply:
@psavol then I'm ok.